app/controllers/api/authentications_controller.rb

```diff
@@ -33,5 +44,14 @@ def build_additional_fields
         :credential_types => ::Authentication.build_credential_options
       }
     end
+
+    def validate_auth_attrs(data)
+      raise 'must supply a manager resource' unless data['manager_resource']
+      attrs = data.dup
+      collection, id = parse_href(data['manager_resource']['href'])
+      raise "#{collection} is not a valid manager resource" unless ::Authentication::MANAGER_TYPES.include?(collection)
```

Comment:
As per @Fryguy the class validation logic is probably better in create_in_provider_queue logic.